### PR TITLE
Add gsuite_group_members resource

### LINKED ISSUE #2: Add group_members array in the group resource
It would be nice to have a `gsuite_group_members` resource.

Currently, if someone adds a user to a group in the Google Admin console, Terraform won't know about it. I would like to have something like this:

```
resource "gsuite_group_members" "devteam" {
  group = "${gsuite_group.devteam.id}"
  owners = [
    "a@b.com",
  ],
  managers = [
    "b@b.com",
  ],
  members = [
    "c@b.com",
  ]
}
```

If this resource is used, `gsuite_group_member` should not be used for the same group.

See https://www.terraform.io/docs/providers/google/r/google_project_services.html and https://www.terraform.io/docs/providers/google/r/google_project_service.html for example

Another option is to have something like this:
```
resource "gsuite_group_members" "access_testing_members" {
  group = "${gsuite_group.access_testing.id}"

  member {
    email = "a@b.com",
    role  = "MEMBER"
  }

  member {
    email = "b@b.com",
    role  = "OWNER"
  }
}
```
What do you think?

### PR DESCRIPTION
**Pull request by psalaberria002**
The inline members are the single source of truth. The resource should control the full set of members.

Fixes #2

### CODE REVIEW

File: gsuite/resource_group_members.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+	config := meta.(*Config)
+
+	members, err := getApiMembers(d.Id(), config)
+
+	if err != nil {
+		return err
+	}
+
+	d.Set("group", d.Id())
+	d.Set("member", membersToCfg(members))
```
Comment (DeviaVir) [inline gsuite/resource_group_members.go]: `gsuite/resource_group_members.go:45:30: cannot use members (type []*admin.Member) as type *admin.Members in argument to membersToCfg`

File: gsuite/resource_group_members.go [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+	cfgMembers := resourceMembers(d)
+
+	// Get members from API
+	apiMembers, err := getApiMembers(gid, config)
+	if err != nil {
+		return gid, fmt.Errorf("Error updating memberships: %v", err)
+	}
+	// This call removes any members that aren't defined in cfgMembers,
+	// and adds all of those that are
+	err = reconcileMembers(d, cfgMembers, membersToCfg(apiMembers), config, gid)
```
Comment (DeviaVir) [inline gsuite/resource_group_members.go]: `gsuite/resource_group_members.go:127:52: cannot use apiMembers (type []*admin.Member) as type *admin.Members in argument to membersToCfg`

File: gsuite/resource_group_members.go [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+	return nil
+}
+
+// Retrieve a group's members from the API
+func getApiMembers(gid string, config *Config) ([]*directory.Member, error) {
+	// Get members from the API
+	groupMembers := make([]*directory.Member, 0)
+	token := ""
+	for paginate := true; paginate; {
+		membersResponse, err := config.directory.Members.List(gid).PageToken(token).Do()
```
Comment (DeviaVir) [inline gsuite/resource_group_members.go]: Do you think this is a sane way to add `retry` logic here?
```
var membersResponse *directory.Members
var err error
err = retry(func() error {
	membersResponse, err = config.directory.Members.List(gid).PageToken(token).Do()
	return err
})
```

File: gsuite/util.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	for k, v := range a {
+		merged[k] = v
+	}
+
+	for k, v := range b {
+		merged[k] = v
+	}
+
+	return merged
+}
```
Comment (DeviaVir) [inline gsuite/util.go]: I'd add this to `utils.go` instead

File: gsuite/resource_group_members.go [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+	for _, raw_member := range d.Get("member").(*schema.Set).List() {
+		member := raw_member.(map[string]interface{})
+		deleteMember(member["email"].(string), d.Id(), config)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func membersToCfg(members *directory.Members) []map[string]interface{} {
```
Comment (DeviaVir) [inline gsuite/resource_group_members.go]:
```
func membersToCfg(members []*directory.Member) []map[string]interface{} {
	if members == nil {
		return nil
	}

	finalMembers := make([]map[string]interface{}, 0, len(members))

	for _, m := range members {
		finalMembers = append(finalMembers, map[string]interface{}{
			"email":  m.Email,
			"etag":   m.Etag,
			"kind":   m.Kind,
			"status": m.Status,
			"type":   m.Type,
			"role":   m.Role,
		})
	}

	return finalMembers
}
```
I think that's how this should look like?

**Comment by DeviaVir** [comment]
Another quite interesting "gotcha":

```
* gsuite_group_members.testing_team_members: Error updating memberships: Error checking hasmember: googleapi: Error 400: Missing required field: memberKey, required
```

is returned when you use an email address of a user that **does not exist**; massively misleading error message.  Should we try and catch it and give a hint?

File: examples/group-members/team.tf [hunk 6]
```diff
@@ -0,0 +1,19 @@
+resource "gsuite_group" "testing_team" {
+  email       = "testing-team@xxx.com"
+  name        = "testing-teamg@xxx.com"
```
Comment (DeviaVir) [inline examples/group-members/team.tf]: `teamg` -- typo?

File: gsuite/resource_group_members.go (hunk 5 above)
Comment (DeviaVir) [reply gsuite/resource_group_members.go]: After applying this fix I can `make` and install, and it works successfully :+1:

Comment (psalaberria002) [reply gsuite/resource_group_members.go]: Yes, you are right. I made changes without trying to compile it again :(

**Comment by psalaberria002** [comment]
Thanks for the review. I will have a look at the error related to `hasmember`.

**Pull request merged**